accounts-db/src/cache_hash_data.rs

```diff
@@ -192,7 +193,8 @@ impl CacheHashDataFile {
 pub(crate) struct CacheHashData {
     cache_dir: PathBuf,
     pre_existing_cache_files: Arc<Mutex<HashSet<PathBuf>>>,
-    should_delete_old_cache_files_on_drop: bool,
+    /// Decides which old cache files to delete.  See `delete_old_cache_files()` for more info.
+    storages_start_slot: Option<Slot>,
```

Comment:
Maybe use a more descriptive Enum instead of generic option to make the meaning that delete more explictly.
``` 
enum DeletePolicy {
  All,
  OnAndAfter(Slot),
}
```

Doesn't have to do in this PR. Can wailt for a follow up PR.